cmake/partition_manager.cmake

```diff
@@ -47,19 +59,61 @@ if(PM_IMAGES OR (EXISTS ${static_configuration_file}))
   # Add subsys defined pm.yml to the input_files
   list(APPEND input_files ${PM_SUBSYS_PREPROCESSED})
 
+
+  if (DEFINED CONFIG_SOC_NRF9160)
+    # See nRF9160 Product Specification, chapter "UICR"
+    set(otp_start_addr "0xff8108")
+    set(otp_size 756) # 189 * 4
+  elseif (DEFINED CONFIG_SOC_NRF5340_CPUAPP)
+    # See nRF5340 Product Specification, chapter Application Core -> ... "UICR"
+    set(otp_start_addr "0xff8100")
+    set(otp_size 764)  # 191 * 4
+  endif()
+
+  math(EXPR flash_size "${CONFIG_FLASH_SIZE} * 1024" OUTPUT_FORMAT HEXADECIMAL)
+
+  if (CONFIG_SOC_NRF9160 OR CONFIG_SOC_NRF5340_CPUAPP)
+    add_region(
+      otp
+      ${otp_size}
+      ${otp_start_addr}
+      start_to_end
+      )
+  endif()
```

Comment:
Let's encode this information into Kconfig instead.

When porting to a new SOC. We want to have as few places as possible that SOC-specific
information is encoded.

If this information was encoded in the soc directory it would be easier for the next port
to be copy and modify these values.

```
if(CONFIG_OTP_FLASH) # Or a better name
    add_region(
      otp
      CONFIG_OTP_FLASH_SIZE
      CONFIG_OTP_FLASH_ADDRESS
      start_to_end
      )
endif()
```

Reply:
Upstream is unlikely to accept this encoded as Kconfig. And encoding this in DT is expected to be too much work for this minor simplification.